TourMate/TourMate/Frontend/Common/Copyable.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  Copyable.swift
+//  TourMate
+//
+//  Created by Keane Chan on 9/4/22.
+//
+
+protocol Copyable {
+    init()
+}
+
+extension Copyable {
+    func copy() -> Self {
+        Self()
```

Comment:
Just checking my understanding, the `copy()` method is needed so that the services are passed by reference?

A little confused here because the all the services are classes so i assume any instance are already reference type

Reply:
Copy method is to ensure that we duplicate the service before passing into the VM. I think structs might work, but changing to structs means we cannot use the delegate pattern anymore, which makes it a little troublesome.